File: EP3/MemoryManager.cpp
#include <cstdio>
#include <iostream>
#include <fstream>
#include <sstream>
#include <vector>
#include <string>

using namespace std;

int main() {
    int total, virt, s, p;
    cout << "hello";
    ifstream file("trace.txt");
    string line, word;

    file >> total >> virt >> s >> p;

    while (getline(file, line)) {
        vector<string> tokens;
        int t0, tf, b, pi, ti;
        string nome;
        istringstream iss(line);

        while (iss >> word)
            tokens.push_back(word);

        t0 = stoi(tokens[0]);
        tf = stoi(tokens[1]);
        nome = tokens[2];
        b = stoi(tokens[3]);


        for (int i = 4; i < tokens.size(); i += 2) {
            pi = stoi(tokens[i]);
            ti = stoi(tokens[i+1]);
        }
    }
}

File: EP3/MemoryManager.py


with open("trace.txt") as file:
    line = file.readline()
    words = line.split()

    total = words[0]
    virt = words[1]
    s = words[2]
    p = words[3]

    print total, virt, s, p, "\n"

    for line in file:
        words = line.split()
        t0 = int(words[0])
        tf = int(words[1])
        nome = words[2]
        b = int(words[3])
        print t0, tf, nome, b
        for i in xrange(4, len(words), 2):
            pi = int(words[i])
            ti = int(words[i+1])
            print pi, ti
